Development/src/pages/devices/ChannelMappingMatrix.js

```diff
@@ -0,0 +1,514 @@
+import { useState } from 'react';
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    Tooltip,
+    Typography,
+} from '@material-ui/core';
+import {
+    ReferenceField,
+    ReferenceManyField,
+    SingleFieldList,
+} from 'react-admin';
+import get from 'lodash/get';
+import ChipConditionalLabel from '../../components/ChipConditionalLabel';
+import MappingButton from '../../components/MappingButton';
+import CollapseButton from '../../components/CollapseButton';
+
+const get_output_caps = (output_item, io) => {
+    return (
+        <>
+            <Typography variant="body1">{'Description:'}</Typography>
+            {output_item.properties.description}
+            <Typography variant="body1">{'Routable inputs:'}</Typography>
+            {output_item.caps.routable_inputs !== null
+                ? output_item.caps.routable_inputs
+                      .map(input_name => {
+                          return input_name === null
+                              ? 'MUTE'
+                              : get(io, `inputs.${input_name}.properties.name`);
+                      })
+                      .join(', ')
+                : 'null'}
+        </>
+    );
+};
+
+const get_input_caps = input_item => {
+    return (
+        <>
+            <Typography variant="body1">{'Description:'}</Typography>
+            {get(input_item, `properties.description`)}
+            <Typography variant="body1">{'Block size:'}</Typography>
+            {get(input_item, `caps.block_size`)}
+            <Typography variant="body1">{'Reordering:'}</Typography>
+            {String(get(input_item, `caps.reordering`))}
+        </>
+    );
+};
+
+const get_source_tooltip = output_item => {
+    return (
+        <>
+            <Typography variant="body1">{'Flows:'} </Typography>
+            <div>
+                <ReferenceManyField
+                    record={output_item}
+                    basePath="/flows"
+                    label="Flows"
+                    source="source_id"
+                    reference="flows"
+                    target="source_id"
+                    link="show"
+                >
+                    <div
```

Comment:
Nice work on resolving the browser console noise, @orfar1994. I have one concern, about the use of channel labels as unique keys, since I don't believe the IS-08 schema requires that the labels are unique within the input or output. Maybe better to use channel index instead? (I think it's only the most local scope in which React requires the keys to be unique.)

Reply:
React requires the keys to be unique on each map function. what do you think about input ID/output ID/channel index as a unique key?

Reply:
I think therefore only necessary to use channel index since each call to `map` only goes over a single input or output's channels. I've pushed this change, see what you think :-)